server/schedulers/balance_plan.go

```diff
@@ -74,11 +74,20 @@ func (p *balanceSchedulerPlan) GetResource(step int) uint64 {
 	}
 	switch step {
 	case pickSource:
-		return p.source.GetID()
+		if p.source != nil {
```

Comment:
why  change it ? And when the p.source is nil?

Reply:
When the top-score store is as target for `ScoreDisallowed`, I was worried about the potential misusage that would cause panic, but not so far. I will add a comment